x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/DataFrameDataExtractor.java

```diff
@@ -155,22 +154,21 @@ public void preview(ActionListener<List<Row>> listener) {
                     return;
                 }
 
-                List<Row> rows = new ArrayList<>(searchResponse.getHits().getHits().length);
+                List<PreviewRow> rows = new ArrayList<>(searchResponse.getHits().getHits().length);
                 for (SearchHit hit : searchResponse.getHits().getHits()) {
-                    var unpooled = hit.asUnpooled();
```

Comment:
@original-brownbear is ok to stop unpooling here as we are not adding the SearchHit to the  row?

Reply:
Scratch that, I decided to reuse the current logic

Reply:
Oh sorry was too slow. If you don't escape any value and process it right away, yes no need to retain a reference or unpool :) That would be the optimal thing if possible. That said, will do my best to review shortly :)

Reply:
No worries, I think I got why you were unpooling and with this change it can be removed for this case.